Source/Provers/SMTLib/SMTLibProcessTheoremProver.cs

```diff
@@ -1108,7 +1108,7 @@ protected void AddTypeDecl(string decl)
 
     ////////////////////////////////////////////////////////////////////////////
 
-    private static string _backgroundPredicates;
+    private string _backgroundPredicates;
```

Comment:
Is this fixing the bug you mention? Can you cover it with a test so that it won't be repeated again?

Reply:
I just tried doing that but to enable defining the test without a lot of code requires some more refactoring in options related code that I'd do as part of another PR. 
Also, it would be a test that closely follows the implementation of `InitializeGlobalInformation` so I don't think it would be too interesting.
Also, without this change the current test suite fails, although it requires executing a combination of existing tests.
Also, having a static field like this was is a smell and I don't think we'll bring that back anytime soon.

Can we leave this test out for now?

Reply:
Ok for the explanation. Could you then please move this field with the other fields at the beginning, so that at least it's obvious why it should not be static. Thanks :-)